server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java

```diff
@@ -22,27 +22,29 @@
 
     String component();
 
-    HealthIndicatorResult calculate();
+    HealthIndicatorResult calculate(boolean calculateDetails);
 
     /**
      * This method creates a HealthIndicatorResult with the given information. Note that it sorts the impacts by severity (the lower the
      * number the higher the severity), and only keeps the 3 highest-severity impacts.
      * @param status The status of the result
      * @param summary The summary used in the result
      * @param details The details used in the result
+     * @param showDetails Whether the details ought to appear in the returned result
      * @param impacts A collection of impacts. Only the 3 highest severity impacts are used in the result
      * @return A HealthIndicatorResult built from the given information
      */
     default HealthIndicatorResult createIndicator(
         HealthStatus status,
         String summary,
         HealthIndicatorDetails details,
+        boolean showDetails,
```

Comment:
Similarly - I think this is not needed as `calculate` already returns EMPTY if it's not calculating it